skrub/_agg_joiner.py

```diff
@@ -0,0 +1,597 @@
+"""
+Implement AggJoiner and AggTarget to join a main table to its auxiliary tables,
+with one-to-many relationships.
+
+Both classes aggregate the auxiliary tables first, then join these grouped
+tables with the base table.
+"""
+from copy import deepcopy
+from typing import Iterable
+
+import numpy as np
+from numpy.typing import ArrayLike
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.utils.multiclass import type_of_target
+from sklearn.utils.validation import check_is_fitted
+
+from skrub._utils import atleast_1d_or_none, atleast_2d_or_none
+from skrub.dataframe import DataFrameLike, SeriesLike
+from skrub.dataframe._namespace import get_df_namespace
+
+NUM_OPERATIONS = ["sum", "mean", "std", "min", "max", "hist", "value_counts"]
+CATEG_OPERATIONS = ["mode", "count", "value_counts"]
```

Comment:
if we use "count" we end up with the (same) count for all categorical operations? also we can't have the count if our table has only numerical columns?

Reply:
are "hist" and "value_counts" the same?

Reply:
Not exactly. Value counts does't accept any argument while hist does. Internaly, we use value counts to perform hist.

Reply:
ok thanks. and IIUC the user passes all the operations name and then skrub splits them into numerical and categorical. for example this means there is no way to ask for value_counts for numerical columns but not categorical ones or vice-versa?

Reply:
Yes, correct. I agree that this feels limiting. 

One idea could be to ask for list first, but to also enable tweaking operations using a dict. 

Do you have suggestions?

Reply:
> if we use "count" we end up with the (same) count for all categorical operations? also we can't have the count if our table has only numerical columns?

You're right. There's probably room for improvement here.

Would you consider running count only once, as an valid operation for both numerical and categorical columns?

Reply:
yes using a dict can be good, or we could have a `numerical_operations` and a `categorical_operations` separate parameters

Reply:
Probably an even better option!

Reply:
> Would you consider running count only once, as an valid operation for both numerical and categorical columns?

yes I guess we could make a special case for count, and include just one count column whenever it appears in numerical ops, categorical ops, or both. or as it is cheap to compute and quite generic we could just add it all the time and remove it from the options. but maybe that's too much of a detail and we can leave that for later

Reply:
I've already created a meta-issue to keep track of our subsequent todos, if you want to add some inputs :) https://github.com/skrub-data/skrub/issues/734